File: data_structure/recursion/example/example1.py

def reverseString(str1) :
    """
    用递归的方法反转字符串
    :param list1: 字符串也数组的形式给出
    :return:
    """
    if not str1 :
        return None
    if len(str1) == 1 :
        return str1[0]
    return reverseString(str1[1:]) + str1[0]


if __name__ == "__main__" :
    str1 = "abcdefg"
    a = reverseString(str1)
    

File: data_structure/recursion/example/example1.go
package main

import "fmt"

func main() {

	a := "abcdefg"
	c := reverseString(a)
	fmt.Println(c)
}

// 字符串递归调用进行反转
func reverseString(str1 string) string {
	if str1 == "" {
		return ""
	}
	if len(str1) == 1 {
		return str1
	}
	result := reverseString(str1[1:]) + string(str1[0])
	return result

}
